tests/components/template/test_vacuum.py

```diff
@@ -44,199 +40,154 @@ async def test_missing_optional_config(hass, calls):
-
-    _verify(hass, STATE_UNKNOWN, None)
-
-    hass.states.async_set(_STATE_INPUT_SELECT, STATE_CLEANING)
-    hass.states.async_set(_BATTERY_LEVEL_INPUT_NUMBER, 100)
-    await hass.async_block_till_done()
-
-    _verify(hass, STATE_CLEANING, 100)
-
-
-async def test_templates_with_valid_values(hass, calls):
-    """Test templates with valid values."""
-    with assert_setup_component(1, "vacuum"):
-        assert await setup.async_setup_component(
-            hass,
+        ),
+        (
+            1,
+            1,
+            STATE_UNKNOWN,
+            None,
             "vacuum",
             {
                 "vacuum": {
                     "platform": "template",
                     "vacuums": {
                         "test_vacuum": {
-                            "value_template": "{{ 'cleaning' }}",
-                            "battery_level_template": "{{ 100 }}",
+                            "value_template": "{{ 'abc' }}",
+                            "battery_level_template": "{{ 101 }}",
                             "start": {"service": "script.vacuum_start"},
                         }
                     },
                 }
             },
-        )
-
-    await hass.async_block_till_done()
-    await hass.async_start()
-    await hass.async_block_till_done()
-
-    _verify(hass, STATE_CLEANING, 100)
-
-
-async def test_templates_invalid_values(hass, calls):
-    """Test templates with invalid values."""
-    with assert_setup_component(1, "vacuum"):
-        assert await setup.async_setup_component(
-            hass,
+        ),
+        (
+            1,
+            1,
+            STATE_UNKNOWN,
+            None,
             "vacuum",
             {
                 "vacuum": {
                     "platform": "template",
                     "vacuums": {
                         "test_vacuum": {
-                            "value_template": "{{ 'abc' }}",
-                            "battery_level_template": "{{ 101 }}",
+                            "value_template": "{{ this_function_does_not_exist() }}",
+                            "battery_level_template": "{{ this_function_does_not_exist() }}",
+                            "fan_speed_template": "{{ this_function_does_not_exist() }}",
                             "start": {"service": "script.vacuum_start"},
                         }
                     },
                 }
             },
-        )
-
-    await hass.async_block_till_done()
-    await hass.async_start()
-    await hass.async_block_till_done()
-
-    _verify(hass, STATE_UNKNOWN, None)
-
-
-async def test_invalid_templates(hass, calls):
-    """Test invalid templates."""
-    with assert_setup_component(1, "vacuum"):
-        assert await setup.async_setup_component(
-            hass,
+        ),
+    ],
+)
+async def test_configs(hass, length, parm1, parm2, start_ha):
+    """Test: configs."""
+    assert len(hass.states.async_all()) == length
+    if length:
```

Comment:
Can we change it so that we pass a list of entity ids to verify instead of `length`? That way we can avoid this if check and just iterate the list of entity ids. We generally don't want any if/else logic in tests.

Reply:
Solved, the solution however was to split it into 2 tests: valid/invalid configs.